test/specs/components/action.spec.jsx

```diff
@@ -0,0 +1,541 @@
+import sinon from 'sinon';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import TestUtils from 'react-addons-test-utils';
+
+import { getMockedStore } from 'test/utils/redux';
+import { mockComponent, findRenderedDOMComponentsWithId } from 'test/utils/react';
+
+import StripeCheckout from 'react-stripe-checkout';
+
+import { ActionComponent } from 'components/action.jsx';
+import { LoadingComponent } from 'components/loading.jsx';
+
+const stripeService = require('services/stripe-service');
+const userService = require('services/user-service');
+
+const AppStore = require('stores/app-store');
+
+const store = AppStore.store;
+
+function mockStore(s, state = {}) {
+    var mockedStore = getMockedStore(s, state);
+
+    Object.defineProperty(AppStore, 'store', {
+        get: () => {
+            return mockedStore;
+        }
+    });
+
+    return mockedStore;
+}
+
+const sandbox = sinon.sandbox.create();
+
+describe('Action', () => {
+    let component;
+    let componentNode;
+    let store;
+
+    beforeEach(() => {
+        mockComponent(sandbox, StripeCheckout, 'div', {
+            className: 'mockedStripe'
+        });
+        mockComponent(sandbox, LoadingComponent, 'div', {
+            className: 'mockedLoading'
+        });
+
+        sandbox.stub(userService, 'immediateUpdate');
+        userService.immediateUpdate.resolves();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    after(() => {
+        Object.defineProperty(AppStore, 'store', {
```

Comment:
since `store` is not a method of `AppStore` it cannot be stubbed. The mock store above takes care of replacing the value by a mock. This part restores it at the initial value.

I really need to try babel-rewire again.